python/bins.cpp

```diff
@@ -155,11 +150,12 @@ void init_buckets(py::module &m) {
   m.def("bins_data", [](py::object &obj) -> py::object {
     auto &var = obj.cast<Variable &>();
     if (var.dtype() == dtype<bucket<Variable>>)
-      return get_buffer<Variable>(obj);
+      return py::cast(obj.cast<Variable &>().bin_buffer<Variable>());
     if (var.dtype() == dtype<bucket<DataArray>>)
-      return get_buffer<DataArray>(obj);
+      return py::cast(obj.cast<Variable &>().bin_buffer<DataArray>().view());
     if (var.dtype() == dtype<bucket<Dataset>>)
-      return get_buffer<Dataset>(obj);
+      // TODO Provide mechanism for creating sharing view as for DataArray above
```

Comment:
What is missing? It looks like it is sharing the buffer just like for DataArray.

Reply:
Buffers yes, but not the dicts, so if you would insert/remove coords or insert/remove dataset items it would have no effect on the original. Would need to add wrapping shared_ptrs in `Dataset` (and hand-write copy constructors). Nothing difficult.